File: bounded-contexts/java/src/main/java/p1/myshop/services/shipping/DeliveryService.java
package p1.myshop.services.shipping;

import p1.myshop.services.catalog.CatalogService;
import p1.myshop.services.shoppingcart.Cart;

public class DeliveryService {
    private final CatalogService catalogService;

    public DeliveryService(CatalogService catalogService) {
        this.catalogService = catalogService;
    }

    public double calculateOrderWeight(Cart cart) {
        return cart.items().stream()
                .mapToDouble(cartItem -> cartItem.quantity.quantity() * catalogService.loadItem(cartItem.itemId).weight())
                .sum();
    }
}

File: bounded-contexts/node/src/services/shipping/delivery.service.ts
import { Cart } from '../../entities/cart'
import { CatalogService } from '../catalog/catalog.service'
import { Inject } from '@nestjs/common'

export class DeliveryService {
  readonly #catalogService: CatalogService

  constructor(@Inject(CatalogService) catalogService: CatalogService) {
    this.#catalogService = catalogService
  }

  calculateOrderWeight(cart: Cart) {
    return cart
      .items()
      .map(
        (cartItem) =>
          cartItem.quantity.quantity() *
          this.#catalogService.loadItem(cartItem.itemId).weight,
      )
      .reduce((sum, weight) => sum + weight, 0)
  }
}
